include/rapidcheck/detail/Unicode.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace rc {
+namespace detail {
+
+template<typename T, typename RandomType>
+T generateCodePoint(rc::detail::BitStream<RandomType>& stream)
+{
+	static_assert(sizeof T >= 3, 
+		"Code points can only be stored in types at leeast three bytes large.");
+
+	// Note, this algorithm is designed to provide
+	// good values for UTF8 encoding but can be
+	// used to generate any Unicode character
+	int maxBytes = 1;
+
+	T codepoint;
+	while (maxBytes < 4)
+	{
+		bool increase = stream.next<bool>();
+		if (!increase)
+		{
+			break;
+		}
+		maxBytes += 1;
+	}
+	int noBits;
+	switch (maxBytes)
+	{
+	case 1:
+		noBits = 7;
+		break;
+	case 2:
+		noBits = 11;
+		break;
+	case 3:
+		noBits = 16;
+		break;
+	default:
+		noBits = 20;
+		// Actually 21, put the first bit
+		// needs to be specially handled
+		// to not exceed the valid
+		// value range for codepoints
+		bool highestBit = stream.next<bool>();
+		if (highestBit)
+		{
+			return 0x100000 | stream.next<T>(16);
+		}
+
+	}
+
+	do
+	{
+		codepoint = stream.next<T>(noBits);
+	} while (codepoint == 0);
+	return codepoint;
+}
+
+template<typename T, typename Y>
+T makeCharacterUtf8(Y codepoint)
```

Comment:
Same here I suppose. UTF-8 will probably never be anything other than sequences of 8-bit chars so `std::string` will probably suffice here.

Just make sure that there's no allocation going on by returning a `std::string` but most implementations of `std::string` should have small-string optimization that works for UTF-8 code units at least.